golang/golangci.yaml

```diff
@@ -0,0 +1,253 @@
+    # To check which checks are enabled run `GL_DEBUG=gocritic golangci-lint run`
+    # By default list of stable checks is used.
+    # Which checks should be disabled; can't be combined with 'enabled-checks'; default is empty
+    disabled-checks:
+      - exitAfterDefer
+      - ifElseChain
+      - elseif
+    # Enable multiple checks by tags, run `GL_DEBUG=gocritic golangci-lint run` to see all tags and checks.
+    # Empty list by default. See https://github.com/go-critic/go-critic#usage -> section "Tags".
+    enabled-tags:
+      - diagnostic
+    #      - style
+    #      - performance
+    #      - experimental
+    #      - opinionated
+    #    disabled-tags:
+    #      - experimental
+    settings: # settings passed to gocritic
+  #      captLocal: # must be valid enabled check name
+  #        paramsOnly: true
+  #      rangeValCopy:
+  #        sizeThreshold: 32
+  gosec:
+    excludes:
+      - G112 # Potential slowloris attack
+  misspell:
+    # Correct spellings using locale preferences for US or UK.
+    # Default is to use a neutral variety of English.
+    # Setting locale to US will correct the British spelling of 'colour' to 'color'.
+    locale: US
+    ignore-words:
+      - cancelled
+  revive:
+    rules:
+      - name: struct-tag
+        arguments: [ "json,inline" ]
+      - name: var-naming
+        arguments:
+          - [ "ID" ,"UUID" ] # Ignore list. The items in this list does not need to follow the rule of initialisms. https://go.dev/wiki/CodeReviewComments#initialisms
+          - ["GRPC"] # Deny list. Add custom initialisms to the linter.
+      - name: redundant-import-alias
+      - name: comment-spacings
+      - name: exported
+        arguments:
+          - disableStutteringCheck
+      - name: package-comments
+  unused:
+    # Mark all struct fields that have been written to as used.
+    # Default: true
+    field-writes-are-uses: true
+    # Treat IncDec statement (e.g. `i++` or `i--`) as both read and write operation instead of just write.
+    # Default: false
+    post-statements-are-reads: false
+    # Mark all exported identifiers as used.
+    # Default: true
+    exported-is-used: true
+    # Mark all exported fields as used.
+    # default: true
+    exported-fields-are-used: false
+    # Mark all function parameters as used.
+    # default: true
+    parameters-are-used: true
+    # Mark all local variables as used.
+    # default: true
+    local-variables-are-used: true
+    # Mark all identifiers inside generated files as used.
+    # Default: true
+    generated-is-used: true
+issues:
+  # Maximum issues count per one linter. Set to 0 to disable. Default is 50.
+  max-per-linter: 0
+  # Maximum count of issues with the same text. Set to 0 to disable. Default is 3.
+  max-same-issues: 0
+  include:
+    - EXC0013
+    - EXC0014
+  exclude-dirs:
+    - clients/kubernetes/types # Exclude the kubernetes client types copied from external sources
```

Comment:
Does these directory specific for a project ?

Reply:
Yes

Reply:
Then we can't have this as the common rule?

Does go linters allows to merge two rules files? So we can have one with the common rules here and each project can maintain project specific ones in each repos ?